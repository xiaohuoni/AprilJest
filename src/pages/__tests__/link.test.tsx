import React from 'react';
import Link from '../link';
import renderer from 'react-test-renderer';

test('Link changes the class when hovered', () => {
  const component = renderer.create(<Link page="http://alitajs.com">Alita</Link>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree!.props.onMouseEnter();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree!.props.onMouseLeave();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
