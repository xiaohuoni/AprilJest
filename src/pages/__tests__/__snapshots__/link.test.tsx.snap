// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="http://alitajs.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Alita
  123
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="http://alitajs.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Alita
  123
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="http://alitajs.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Alita
  123
</a>
`;
